src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -469,25 +488,34 @@ public int Length
                     throw new ArgumentOutOfRangeException(nameof(value), value, SR.ArgumentOutOfRange_NeedNonNegNum);
                 }
 
-                int newints = GetArrayLength(value, BitsPerInt32);
-                if (newints > m_array.Length || newints + _ShrinkThreshold < m_array.Length)
+                int newints = GetIntegerArrayLengthFromBitLength(value);
+
+                // grow or shrink (if wasting more than _ShrinkThreshold ints)
+                if (newints > m_array.Length)
                 {
-                    // grow or shrink (if wasting more than _ShrinkThreshold ints)
-                    Array.Resize(ref m_array, newints);
+                    var newArray = new int[newints];
+                    m_array.AsSpan().CopyTo(newArray);
+                    m_array = newArray;
+                }
+                else if (newints + _ShrinkThreshold < m_array.Length)
+                {
+                    var newArray = new int[newints];
+                    m_array.AsSpan(0, newints).CopyTo(newArray);
+                    m_array = newArray;
```

Comment:
Why avoid the Array.Resize call?  Isn't this relatively rare, such that it'd be better to just reuse the existing implementation?

Reply:
You are right, but in the current implementation, where we essentially grow by one, it isn't that rare. I could implement it by doubling the size instead, which would make it super rare. But, I suppose depth growing by 32 is actually rare, regardless (since increase by 1 = 32 new bits).

Reply:
If growing by one makes it so that it isn't rare, then you shouldn't be growing by 1 ;)  This path allocates an array and copies all of the data, so you should only be doing that rarely and choosing a growth pattern that, for this use case, makes it rare.  And either way, the extra method call for the resizing to be encapsulated and avoid the code duplication shouldn't be impactful.

Reply:
I don't think doubling is necessary.  The pathological case is something like setting bit 2047, then clearing it (dropping everything), then setting it again, and doubling doesn't help that.  (Changing a grow/shrink algorithm really requires usage analysis)

And I agree with Steve, the method call shouldn't really be considered an issue.